utils/tinymix.c

```diff
@@ -87,57 +122,68 @@ int main(int argc, char **argv)
             break;
         case 'h':
             usage();
+            free(argv_options_list);
             return EXIT_SUCCESS;
         case 'v':
             version();
+            free(argv_options_list);
             return EXIT_SUCCESS;
         case '?':
-            fprintf(stderr, "%s\n", opts.errmsg);
-            return EXIT_FAILURE;
+        default:
+            break;
         }
     }
+    free(argv_options_list);
 
     mixer = mixer_open(card);
     if (!mixer) {
         fprintf(stderr, "Failed to open mixer\n");
         return EXIT_FAILURE;
     }
 
-    cmd = argv[opts.optind];
-    if (cmd == NULL) {
-        fprintf(stderr, "no command specified (see --help)\n");
+    command_position = find_command_position(argc, argv);
```

Comment:
It would be better to reduce the life time of `command_position` and have it declared here instead of at the top of the function.

Reply:
I agree with you, but as I said in the previous comment, do we need to keep C89 standard?